controllers/metal3.io/host_state_machine.go

```diff
@@ -96,12 +97,13 @@ func (hsm *hostStateMachine) updateHostStateFrom(initialState metal3v1alpha1.Pro
 	if hsm.NextState != initialState {
 
 		// Check if there is a free slot available when trying to
-		// provision an host - if not the action will be delayed.
-		// The check is limited to only the provisioning states to
+		// (de)provision an host - if not the action will be delayed.
+		// The check is limited to only the (de)provisioning states to
 		// avoid putting an excessive pressure on the provisioner
 		switch hsm.NextState {
-		case metal3v1alpha1.StateInspecting, metal3v1alpha1.StateProvisioning:
-			if actionRes := hsm.ensureProvisioningCapacity(info); actionRes != nil {
+		case metal3v1alpha1.StateInspecting, metal3v1alpha1.StateProvisioning,
+			metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
+			if actionRes := hsm.ensureCapacity(info, hsm.NextState); actionRes != nil {
```

Comment:
Not a blocking point, but I was noticing that since this list is now getting bigger, it would be nice to remove the code duplication of this block with the one in[ checkDelayedHost](https://github.com/metal3-io/baremetal-operator/pull/807/files#diff-eeed3703d04e4c23a7d7af8cd0b7931b6b7990f23d826c49bdbc31c5f0a50291R160) - I guess this could also helps in passing the provisioning state

Reply:
@andfasano Thanks for approve, lets do code deduplication as a separate PR.

Reply:
Changes look fine to me, but I'd like to give @dhellmann and @dtantsur the opportunity for a final review

Reply:
Don't block this work on me. I'm not likely to have time to come back and review it closely for a couple of weeks.

Reply:
Same as Doug, unfortunately.